packages/monaco/src/browser/monaco-workspace.ts

```diff
@@ -175,14 +179,20 @@ export class MonacoWorkspace implements lang.Workspace {
     @inject(ProblemManager)
     protected readonly problems: ProblemManager;
 
+    protected _workspaceFolders: WorkspaceFolder[];
+    get workspaceFolders(): WorkspaceFolder[] {
+        return this._workspaceFolders;
+    }
+
     @postConstruct()
-    protected init(): void {
-        this.workspaceService.roots.then(roots => {
-            const rootStat = roots[0];
-            if (rootStat) {
-                this._rootUri = rootStat.uri;
-                this.resolveReady();
-            }
+    protected async init(): Promise<void> {
+        await this.workspaceService.roots.then(roots => {
```

Comment:
Does the `await` look good with the `then` here?

Reply:
i think so, we should not listen to events till the initial state is computed

Reply:
Fixed